pkg/graph/drive.resolvers.go

```diff
@@ -424,6 +457,142 @@ func (r *mutationResolver) CreateSharedDrive(ctx context.Context, name string) (
 	return bucket, nil
 }
 
+// CreateShare is the resolver for the createShare field.
+func (r *mutationResolver) CreateShare(ctx context.Context, input model.CreateShareInput) (*model.ShareUser, error) {
+	currentUser, err := middleware.GetUser(ctx)
+	if err != nil {
+		return nil, nil
+	}
+
+	// Get user
+	var user db.User
+	err = r.DB.NewSelect().Model(&user).Where("id = ?", input.User).Scan(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if we have fileId or bucketId
+	if input.FileID == nil && input.BucketID == nil {
+		return nil, errors.New("fileId or bucketId is required")
+	}
+
+	share := db.Share{
+		SharedWith:     input.User,
+		SharedBy:       currentUser.ID,
+		OrganisationID: currentUser.OrganisationID,
+		Permission:     strings.ToLower(string(input.Permission)),
+	}
+
+	if input.FileID != nil {
+		share.FileID = *input.FileID
+	} else {
+		share.BucketID = *input.BucketID
+	}
+
+	err = r.DB.NewInsert().Model(&share).Returning("*").Scan(ctx)
```

Comment:
Are we before inserting checking if the user has the permission to share this folder/bucket?